src/pybind/mgr/ansible/module.py

```diff
@@ -331,33 +431,44 @@ def remove_osds(self, osd_ids):
                      'ireallymeanit':'yes'}
 
         # Create a new read completion object for execute the playbook
-        ansible_operation = AnsibleReadOperation(client = self.ar_client,
-                                                 playbook = REMOVE_OSD_PLAYBOOK,
-                                                 logger = self.log,
-                                                 result_pattern = "",
-                                                 params = extravars)
+        playbook_operation = PlaybookOperation(client = self.ar_client,
+                                               playbook = REMOVE_OSD_PLAYBOOK,
+                                               logger = self.log,
+                                               result_pattern = "",
+                                               params = extravars)
 
         # Filter to get the result
-        ansible_operation.process_output = process_playbook_result
-        ansible_operation.event_filter = "playbook_on_stats"
+        playbook_operation.process_output = process_playbook_result
+        playbook_operation.event_filter = "playbook_on_stats"
 
         # Execute the playbook
-        self._launch_operation(ansible_operation)
+        self._launch_playbook(playbook_operation)
+
+        return playbook_operation
+
+
+    def get_hosts(self):
+        """Provides a list Inventory nodes
+        """
+
+        host_ls_op = ARSOperation(self.ar_client, self.log, URL_GET_HOSTS)
 
-        return ansible_operation
+        host_ls_op.process_output = process_hosts_ls
```

Comment:
why not passing `process_hosts_ls` as argument for ARSOperation? e.g.

```python
ARSOperation(self.ar_client, self.log, URL_GET_HOSTS, process_hosts_ls)
```

Reply:
My first thought when implementing "PlaybookOperation" was that this setting sometimes could be optional ... in that moment i only had this "PlaybookOperation" class, ant it uses 6 parameters to init... so it seemed to me better no to include this setting as parameter of the init method.

In order to implement the list of hosts, it has been needed to add another "dedicated" completion object for doing operations with the Ansible Runer Service directly ( not to execute a playbook).

This object ifs the "ARSOperation" , and in this case, the number of init parameters is only three, and as you said, it could seem sensible to include the "process_host_ls" in the init method.
But, "ARSOperation" and "PlaybookOperation" are sisters/brothers ( in spanish a class is female:-) , so i think that to use "process_host_ls" in both classes in the same way is more important .